clevertap-core/src/main/java/com/clevertap/android/sdk/inapp/ImpressionManager.kt

```diff
@@ -0,0 +1,216 @@
+package com.clevertap.android.sdk.inapp
+
+import com.clevertap.android.sdk.inapp.store.preference.StoreRegistry
+import com.clevertap.android.sdk.utils.Clock
+import java.util.Calendar
+import java.util.Date
+import java.util.Locale
+import java.util.concurrent.TimeUnit
+
+/**
+ * Provides functionality for tracking and managing impressions for various campaigns.
+ *
+ * @property storeRegistry A storage manager responsible for storing and retrieving impression data.
+ * @property clock           An optional Clock implementation for handling time-related operations.
+ * @property locale          An optional Locale specifying the locale to use for date and time calculations.
+ */
+class ImpressionManager @JvmOverloads constructor(
+    private val storeRegistry: StoreRegistry,
+    private val clock: Clock = Clock.SYSTEM,
+    private val locale: Locale = Locale.US,
```

Comment:
why are we using Locale.US
I hope we do not display any time with this

Reply:
@shivamsharma2710 confirm with @nzagorchev and backend

Reply:
discussed in standup